libraries/Microsoft.Bot.Builder/Streaming/StreamingRequestHandler.cs

```diff
@@ -371,7 +368,25 @@ private void Server_Disconnected(object sender, DisconnectedEventArgs e)
 
         private async Task ReconnectAsync(IDictionary<string, string> requestHeaders = null)
         {
-            // TODO: Need to get authentication headers from the httpClient on the adapter.
+            // The ServiceUrl of a streaming connection follows the pattern "urn:[ServiceName]:[Protocol]:[ChannelName]".
+            var streamingUrnPattern = new Regex("urn:(.+?:){2}.+");
+            string[] urnSections;
+            try
+            {
+                urnSections = streamingUrnPattern.Split(ServiceUrl);
+            }
+            catch (Exception ex)
```

Comment:
Should we catch regex specific exception?

Reply:
This is also serving as a null check now. If we want to handle them differently I'll add an explicit null/empty/whitespace check first, but in either case the information is going to point to the ServiceURL on the request handler not being set to an appropriate value.